app.py

```diff
@@ -1,26 +1,402 @@
+        '-l', '--local',
+        help="Pass to attempt scraping from local files.\n",
+        action='store_true',
+        dest='fresh_search',
+        default=False)
+    parser.add_argument(
+        '-s', '--save',
+        help="Pass to save plots locally\n(if omitted will show plots instead)\n",
+        action='store_true',
+        dest='save_plot',
+        default=False)
+    parser.add_argument(
+        '-c', '--count',
+        help="The amount of tweets to search for.\n",
+        type=int,
+        default=300,
+        dest='tweet_count')
+    parser.add_argument(
+        '-d', '--date',
+        help="The date range (yyyy-mm-dd) to search for. \nEXAMPLE: '2020-05-01 2020-05-05'.\n",
+        nargs=2,
+        type=_restricted_dates,
+        default=_default_dates(),
+        dest='date')
+    parser.add_argument(
+        '-se', '--sentiment',
+        help="Ignore specific sentiment.\nVALUES=[Positive, Negative, Uncertain]\n",
+        type=_restricted_sentiment,
+        dest='remove_sentiment')
+    parser.add_argument(
+        '-sh',
+        help="Filter result data by specific hashtags.\nEXAMPLE: '#Trump #Biden'\n",
+        type=_restricted_search_hashtags,
+        nargs='+',
+        dest='search_hashtags')
+    parser.add_argument(
+        '-sm',
+        help="Filter result data by specific mentions.\nEXAMPLE: '@JoeBiden @folketinget'\n",
+        type=_restricted_search_mentions,
+        nargs='+',
+        dest='search_mentions')
+    parser.add_argument(
+        '-su',
+        help="Filter result data by specific URLs.\nEXAMPLE: 'https://pic.twitter.com/'\n",
+        type=str,
+        nargs='+',
+        dest='search_urls')
+    parser.add_argument(
+        '-cl',
+        help="The lower float value for determining if a sentiment is deemed uncertain.\nEXAMPLE: 0.15.\nVALUES: [0.00-1.00]",
+        type=_restricted_float,
+        default=0.25,
+        metavar='ADVANCED: certainty low',
+        dest='certainty_low')
+    parser.add_argument(
+        '-ch',
+        help="The higher float value for determining if a sentiment is deemed uncertain.\nEXAMPLE: 0.65.\nVALUES: [0.00-1.00]",
+        type=_restricted_float,
+        default=0.75,
+        metavar='ADVANCED: certainty high',
+        dest='certainty_high')
+    # endregion
+    # turn Namespace object into usable dict
+    args_dict = vars(parser.parse_args())
+    print('--------------------------------------')
+    print('Received Values:')
+    pprint(args_dict, width=1)
+    print('--------------------------------------')
+
+    # extract items from dict
+    hashtags, plot_type, fresh_search, tweet_count, date, save_plot, remove_sentiment, search_hashtags, search_mentions, search_urls, certainty_low, certainty_high = itemgetter(
+        'hashtags', 'plot_type', 'fresh_search', 'tweet_count', 'date', 'save_plot', 'remove_sentiment', 'search_hashtags', 'search_mentions', 'search_urls', 'certainty_low', 'certainty_high')(args_dict)
```

Comment:
So CLI makes a dict, which you then destrucutre and feed to main method?

Reply:
CLI actually makes a Namespace object which looks like this:  
`Namespace(bar='BAR', foo='1')`
It isn't always easy to grab these values, so the suggested way to grab them is as I have done here. A bit hard to read as we have so many arguments, but it works.

Reply:
https://docs.python.org/dev/library/argparse.html#argparse.Namespace

Perfect, thanks